pkg/apis/core/validation/events.go

```diff
@@ -33,8 +37,83 @@ const (
 	NoteLengthLimit              = 1024
 )
 
-// ValidateEvent makes sure that the event makes sense.
-func ValidateEvent(event *core.Event) field.ErrorList {
+func ValidateEventCreate(event *core.Event, requestVersion schema.GroupVersion) field.ErrorList {
+	// Make sure events always pass legacy validation.
+	legacyErrs := legacyValidateEvent(event)
+	if len(legacyErrs) > 0 {
+		return legacyErrs
```

Comment:
I know that you did what Jordan suggested, but I think we should actually try to return all possible errors instead of returning here.

I think we should change to:
```
allErrs := legacyValidateEvent(event)
if requestVersion == core.SchemeGroupVersion || requestVersion == eventsv1beta1.SchemeGroupVersion {
	// No further validation for backwards compatibility.
	return allErrs
}

allErrs = append(allErrs, ValidateObjectMeta(...))
allErrs = append(allErrs, validateV1EventSpec(event)...)
return allErrs
```

Reply:
Done.